iotilebuild/iotile/build/config/site_scons/trub_script.py

```diff
@@ -27,51 +29,77 @@ def build_update_script(file_name, slot_assignments, os_info=None):
             This file name should end in .trub
         slot_assignments (list of (str, str)): A list of tuples containing
             the slot name and the firmware image that we should use to build
-            our update script.
+            our update script. Optional
         os_info (tuple(int, str)): A tuple of OS version tag and X.Y version
-            number that will be set as part of the OTA script if included.
+            number that will be set as part of the OTA script if included. Optional.
+        sensor_graph (str): Name of sgf file. Optional.
+        app_info (tuple(int, str)): A tuple of App version tag and X.Y version
+            number that will be set as part of the OTA script if included. Optional.
     """
 
     resolver = ProductResolver.Create()
     env = Environment(tools=[])
+    files = []
 
-    slots = [_parse_slot(x[0]) for x in slot_assignments]
-    files = [ensure_image_is_hex(resolver.find_unique("firmware_image", x[1]).full_path) for x in slot_assignments]
+    if slot_assignments is not None:
+        slots = [_parse_slot(x[0]) for x in slot_assignments]
+        files = [ensure_image_is_hex(resolver.find_unique("firmware_image", x[1]).full_path) for x in slot_assignments]
+        env['SLOTS'] = slots
 
-    env['SLOTS'] = slots
     env['OS_INFO'] = os_info
+    env['APP_INFO'] = app_info
+
+    if sensor_graph is not None:
+        files.append(sensor_graph)
+
     env.Command([os.path.join('build', 'output', file_name)], files, action=Action(_build_reflash_script_action, "Building TRUB script at $TARGET"))
 
 
 def _build_reflash_script_action(target, source, env):
-    """Create a TRUB script containing tile and controller reflashes.
+    """Create a TRUB script containing tile and controller reflashes and/or sensorgraph
 
-    All of the files in source must be in intel hex format. This is guaranteed
+    If the app_info is provided, then the final source file will be a sensorgraph.
+    All subsequent files in source must be in intel hex format. This is guaranteed
     by the ensure_image_is_hex call in build_update_script.
     """
 
     out_path = str(target[0])
     source = [str(x) for x in source]
     records = []
-
-    for (controller, slot_id), image_path in zip(env['SLOTS'], source):
-        hex_data = IntelHex(image_path)
-        hex_data.padding = 0xFF
-
-        offset = hex_data.minaddr()
-        bin_data = bytearray(hex_data.tobinarray(offset, hex_data.maxaddr()))
-
-        if controller:
-            record = ReflashControllerRecord(bin_data, offset)
-        else:
-            record = ReflashTileRecord(slot_id, bin_data, offset)
-
-        records.append(record)
-
+    
+    #Update sensorgraph
+    app_info = env['APP_INFO']
+    if app_info is not None:
```

Comment:
This assumes that app_info and sensorgraph will always be set together, even though nothing in the calling function requires that (they are both separate parameters with default values that can be set independently).  If you want to ensure they are set together you'll need add a check in the calling function that throws an exception if `app_info` but no `sensor_graph`.

However, I would guess that it would be better to just allow them both to be set independently and remove the assumption in this function that `app_info is not None` means sensorgraph.  You can use `ENV` to pass whatever additional flags you want to this function.

Reply:
Good point, i'm going to allow them to be set independently.